internal/driver/fetch.go

```diff
@@ -379,6 +381,63 @@ func fetchURL(source string, timeout time.Duration) (io.ReadCloser, error) {
 	return resp.Body, nil
 }
 
+// profileProtoReader takes a path, and using heuristics, will try to convert
+// the file to profile.proto format. It returns a ReadCloser to the
+// profile.proto data; however, if the file contents were unknown or conversion
+// failed, it may still not be a valid profile.proto.
+func profileProtoReader(path string, ui plugin.UI) (io.ReadCloser, error) {
+	sourceFile, openErr := os.Open(path)
+	if openErr != nil {
+		return nil, openErr
+	}
+
+	// If the file is the output of a perf record command, it should begin
+	// with the string PERFILE2.
+	perfHeader := []byte("PERFILE2")
+	actualHeader := make([]byte, len(perfHeader))
+	_, readErr := sourceFile.Read(actualHeader)
+	_, seekErr := sourceFile.Seek(0, 0)
+	if seekErr != nil {
+		return nil, seekErr
+	}
+	if readErr != nil && readErr != io.EOF{
+		return nil, readErr
+	}
+	if bytes.Equal(actualHeader, perfHeader) {
+		sourceFile.Close()
+		profileFilePath, convertErr := convertPerfData(path, ui)
+		if convertErr != nil {
+			return nil, convertErr
+		}
+		profileFile, openErr := os.Open(profileFilePath)
+		if openErr != nil {
+			os.Remove(profileFilePath)
+			return nil, openErr
+		}
+		return DeleteOnClose(profileFile), nil
+	}
+	return sourceFile, nil
+}
+
+// convertPerfData converts the file at path which should be in perf.data format
+// using the perf_to_profile tool and returns the path to a file containing the
+// profile.proto formatted data.
+func convertPerfData(perfPath string, ui plugin.UI) (string, error) {
+	ui.Print(fmt.Sprintf(
+		"Converting %s to a profile.proto... (May take a few minutes)",
+		perfPath))
+	profilePath, err := newTempFilePath("/tmp", "pprof_", ".pb.gz")
+	if err != nil {
+		return "", err
+	}
+	cmd := exec.Command("perf_to_profile", perfPath, profilePath)
```

Comment:
Where is this command available? It looks like it would be similar to https://godoc.org/rsc.io/perf2pprof, but with a different interface.